pkg/controller/node/node_controller.go

```diff
@@ -99,19 +100,18 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcileResult, err
 	}
 
-	// Set a storageos cluster client.
-	if r.stosClient != nil {
-		// Compare the cluster names, generations and UUIDs to check if it's
-		// the same cluster. Update the client if client cluster name,
-		// generation or UID are different from current cluster.
-		if r.stosClient.clusterName != cluster.GetName() ||
-			r.stosClient.clusterGeneration != cluster.GetGeneration() ||
-			r.stosClient.clusterUID != cluster.GetUID() {
-			r.setClientForCluster(cluster)
+	// Compare the cluster names, generations and UUIDs to check if it's
+	// the same cluster. Update the client if client cluster name,
+	// generation or UID are different from current cluster.
+	if r.stosClient == nil ||
+		r.stosClient.clusterName != cluster.GetName() ||
+		r.stosClient.clusterGeneration != cluster.GetGeneration() ||
+		r.stosClient.clusterUID != cluster.GetUID() {
+
+		if err := r.setClientForCluster(cluster); err != nil {
+			log.Println("failed to configure api client:", err)
+			return reconcileResult, err
```

Comment:
This is much better :+1:  return error and requeue the request.
I missed error checking :sweat_smile: